File: Chap6-GraphicsViewFramework/6-2EventPropagationDemo/scene.py
from PySide6.QtWidgets import QGraphicsScene
from PySide6.QtCore import QObject
from PySide6.QtGui import QKeyEvent
from PySide6.QtWidgets import QGraphicsSceneMouseEvent

class Scene(QGraphicsScene):
    """
    A custom graphics scene that demonstrates event handling
    """
    def __init__(self, parent=None):
        super().__init__(parent)

    def keyPressEvent(self, event: QKeyEvent):
        """
        Handle key press events and pass them to the parent class
        """
        print("Scene : KeypressEvent")
        super().keyPressEvent(event)

    def mousePressEvent(self, event: QGraphicsSceneMouseEvent):
        """
        Handle mouse press events and pass them to the parent class
        """
        print(f"Scene : MousePressEvent at : {event.scenePos()}")
        super().mousePressEvent(event)

File: 16-GraphicsViewFramework/02_EventPropagationDemo/scene.cpp
#include "scene.h"
#include <QDebug>
#include <QKeyEvent>
#include <QGraphicsSceneMouseEvent>

Scene::Scene(QObject *parent) : QGraphicsScene(parent)
{

}

void Scene::keyPressEvent(QKeyEvent *event)
{
    qDebug() << "Scene : KeypressEvent : " ;
    QGraphicsScene::keyPressEvent(event);

}

void Scene::mousePressEvent(QGraphicsSceneMouseEvent *event)
{
    qDebug() << "Scene : MousePressEvent at : " << event->scenePos();
    QGraphicsScene::mousePressEvent(event);

}
